curriculum/challenges/english/25-front-end-development/quiz-javascript-higher-order-functions/66edcdd18a4ef8df16e6bb7e.md

```diff
@@ -17,439 +17,482 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Evaluate the output
+
+```js
+[, undefined, 'a', 'b', { 20: 5 }].sort();
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+unsupported elements for an array to be sorted, hence error
 
 ---
 
-Placeholder distractor 2
+Callback not supplied, hence error
 
 ---
 
-Placeholder distractor 3
+```js
+[<1 empty item> , 'a', 'b', undefined,{ '20': 5 } ]
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+[ { '20': 5 }, 'a', 'b', undefined, <1 empty item> ]
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following describes a callback function in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+## A function that is called immediately upon declaration.
 
----
-
-Placeholder distractor 2
+A function that is invoked with a specific context.
 
 ---
 
-Placeholder distractor 3
+A function that returns another function.
 
 #### --answer--
 
-Placeholder answer
+A function passed as an argument to another function, to be executed later.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of using reduce() on an array?
 
 #### --distractors--
 
-Placeholder distractor 1
+A boolean indicating whether any elements meet a condition.
 
 ---
 
-Placeholder distractor 2
+An Array with all elements reduced by specified callback function
 
 ---
 
-Placeholder distractor 3
+An Array of booleans
 
 #### --answer--
 
-Placeholder answer
+A single accumulated value derived from all elements in the array.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does the sort() method behave if no comparison function is provided?
 
 #### --distractors--
 
-Placeholder distractor 1
+It sorts the array as strings based on UTF-16 code units but takes undefined, empty slots and object references the start
 
 ---
 
-Placeholder distractor 2
+It returns an error.
 
 ---
 
-Placeholder distractor 3
+It sorts the array in reverse order.
 
 #### --answer--
 
-Placeholder answer
+It sorts the array as strings based on UTF-16 code units but moves empty slots to the end of the array and object references converts to "[object Object]"
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the some() method in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+To create a new array with the results of a function applied to each element.
 
 ---
 
-Placeholder distractor 2
+To iterate through an array without producing a result.
 
 ---
 
-Placeholder distractor 3
+To reduce an array to a single value based on a callback function.
 
 #### --answer--
 
-Placeholder answer
+To determine if any elements in an array pass a specified test.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is an example of method chaining?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+Math.random();
+```
 
 ---
 
-Placeholder distractor 2
+```js
+array.push(1).pop();
```

Comment:
Isn't this technically also method chaining? Perhaps, question could be changed to specifically ask for correct usage?

Reply:
you did not address this @bharateshwq

Reply:
I think it would be beneficial to clarify that method chaining isn't always applicable just because a method returns a value. The methods involved must actually support chaining.

Reply:
this contradicts an other question where you have that `push` supports chaining

Reply:
From what i understand push returns length of the resulting array as its output, this means the resulting number has toString(), toFixed() and many more functions to be chained similar to most of the functions where array is returned. This options might even educate a student that push returns a number which means .pop() is unsupported.

Reply:
We can resolve this by changing the question a bit. See my suggestion above.